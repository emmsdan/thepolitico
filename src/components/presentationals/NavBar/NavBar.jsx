import React from 'react';
import PropTypes from 'prop-types';

const NavBar = props => {
  const li = [];
  props.pages.forEach((page, id) => {
    li.push(
      <li className={page.className} key={id}>
        <a href={page.url}>{page.title}</a>
      </li>,
    );
  });
  return (
    <nav role="navigation">
      <ul> {li} </ul>
    </nav>
  );
};

NavBar.propTypes = {
  pages: PropTypes.array,
};

export default NavBar;
