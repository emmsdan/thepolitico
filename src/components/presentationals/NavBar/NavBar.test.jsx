import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import NavBar from './NavBar';

let store;
describe('<NavBar />', () => {
  beforeEach(() => {
    const mockStore = configureMockStore();
    store = mockStore({ auth: { isLoading: false } });
  });

  const pages = [
    {
      title: 'home',
      url: './',
      className: '',
    },
    {
      title: 'results',
      url: './results',
      className: '',
    },
  ];
  it('should render correctly', () => {
    const component = shallow(
      <Provider store={store}>
        <NavBar pages={pages} />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
