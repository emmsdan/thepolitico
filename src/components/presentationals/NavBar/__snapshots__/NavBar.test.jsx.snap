// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render correctly 1`] = `
<nav
  role="navigation"
>
  <ul>
     
    <li
      className=""
      key="0"
    >
      <a
        href="./"
      >
        home
      </a>
    </li>
    <li
      className=""
      key="1"
    >
      <a
        href="./results"
      >
        results
      </a>
    </li>
     
  </ul>
</nav>
`;
